api/files.py

```diff
@@ -138,15 +137,18 @@ def _FieldStorage__write(self, line):
                 # Always write fields of type "file" to disk for consistent renaming behavior
                 if self.filename:
                     self.file = self.make_file('')
-
                     self.file.write(self._FieldStorage__file.getvalue())
+                    self.hasher.update(self._FieldStorage__file.getvalue())
                 self._FieldStorage__file = None
+
             self.file.write(line)
+            if hasattr(self, 'hasher'):
```

Comment:
Were you able to track down how this can get into a state where `self.hasher` is not set at this point?

Reply:
In case of the metadata we don't have a file name so we won't create the file and hasher. I'll add a comment to clarify this.